### PR TITLE
Backports 0.20 pr6

### PR DESCRIPTION
**Pull request by Munkybooty**
So i've been struggling trying to get #16908 to build. @UdjinM6 do you know of any good solution, to deal with the timestamp in `getaddressmempool`?

### CODE REVIEW

**Comment by knst** [comment]
> So i've been struggling trying to get https://github.com/bitcoin/bitcoin/pull/16908 to build. @UdjinM6 do you know of any good solution, to deal with the timestamp in getaddressmempool?

@Munkybooty , please, add missing <chrono> header and call count_seconds() in UniValue's related code.
It's enough for my gcc 11, please, tell if you have any extra issues with compilation.
```
$ git diff
diff --git a/src/addressindex.h b/src/addressindex.h
index 773f426de..ea8853a94 100644
--- a/src/addressindex.h
+++ b/src/addressindex.h
@@ -9,6 +9,8 @@
 #include <uint256.h>
 #include <amount.h>
 
+#include <chrono>
+
 struct CMempoolAddressDelta
 {
     std::chrono::seconds time;
diff --git a/src/rpc/misc.cpp b/src/rpc/misc.cpp
index 2b7ff4c1d..cd67c004e 100644
--- a/src/rpc/misc.cpp
+++ b/src/rpc/misc.cpp
@@ -714,7 +714,7 @@ static UniValue getaddressmempool(const JSONRPCRequest& request)
         delta.pushKV("txid", it->first.txhash.GetHex());
         delta.pushKV("index", (int)it->first.index);
         delta.pushKV("satoshis", it->second.amount);
-        delta.pushKV("timestamp", it->second.time);
+        delta.pushKV("timestamp", count_seconds(it->second.time));
         if (it->second.amount < 0) {
             delta.pushKV("prevtxid", it->second.prevhash.GetHex());
             delta.pushKV("prevout", (int)it->second.prevout);

```

**Comment by knst** [comment]
Hm, functional test `mempool_persist.py` is failed, because python script calls real time here `tx_creation_time_lower = int(time.time())` [current real world time], but inside mempool or time are mocked in GetTime() to 2014 year.

failure output:
```
Traceback (most recent call last):
  File "/home/knst/projects/dash-reviews/test/functional/test_framework/test_framework.py", line 233, in main
    self.run_test()
  File "/home/knst/projects/dash-reviews/test/functional/mempool_persist.py", line 79, in run_test
    assert_greater_than_or_equal(tx_creation_time, tx_creation_time_lower)
  File "/home/knst/projects/dash-reviews/test/functional/test_framework/util.py", line 67, in assert_greater_than_or_equal
    raise AssertionError("%s < %s" % (str(thing1), str(thing2)))
AssertionError: 1417744711 < 1656921390
```

It happens, because we set up mocktime in `test/functional/test_framework/test_framework.py` always, but `bitcoin` in functional tests does it only for some specific tests.

You may update functional test like that to fix this failure:
```py
--- a/test/functional/mempool_persist.py
+++ b/test/functional/mempool_persist.py
@@ -57,12 +57,12 @@ class MempoolPersistTest(BitcoinTestFramework):
 
     def run_test(self):
         self.log.debug("Send 5 transactions from node2 (to its own address)")
-        tx_creation_time_lower = int(time.time())
+        tx_creation_time_lower = self.mocktime
         for i in range(5):
             last_txid = self.nodes[2].sendtoaddress(self.nodes[2].getnewaddress(), Decimal("10"))
         node2_balance = self.nodes[2].getbalance()
         self.sync_all()
-        tx_creation_time_higher = int(time.time())
+        tx_creation_time_higher = self.mocktime
```

**Comment by knst** [comment]
Check this one: [make check fix](https://github.com/knst/dash/commit/c6eeee341d06bf6bae72724425d191343941759e) and [Merge #20852](https://github.com/knst/dash/commit/c5ea4d1a0ed56da49094f7d492813d0be10eaa07)

File: src/qt/bitcoingui.cpp [hunk 1]
```diff
@@ -400,8 +400,8 @@ void BitcoinGUI::createActions()
 
     openInfoAction = new QAction(tr("&Information"), this);
     openInfoAction->setStatusTip(tr("Show diagnostic information"));
-    openRPCConsoleAction = new QAction(tr("&Debug console"), this);
-    openRPCConsoleAction->setStatusTip(tr("Open debugging console"));
+    openRPCConsoleAction = new QAction(tr("&Tools window"), this);
```
Comment (UdjinM6) [inline src/qt/bitcoingui.cpp:403]: 17096: (unlike in bitcoin) we have menu items for each tab here, so this change is not applicable imo
```suggestion
    openRPCConsoleAction = new QAction(tr("&Debug console"), this);
```

**Review by UdjinM6 — CHANGES_REQUESTED** [review]
pls see below + 1ae8cd6d33ea3c7105676105b24dc7cd0bfa3801 and d1166f814a8161a2486d78c0b988ce5ae947421e

**Review by UdjinM6 — DISMISSED** [review]
LGTM, utACK

**Review by UdjinM6 — DISMISSED** [review]
re-utACK

**Review by UdjinM6 — CHANGES_REQUESTED** [review]
19208 was recently backported via #5009 (https://github.com/dashpay/dash/pull/5009/commits/566889f5e8e1e811a6c127b99a785660ebc4db1b) so `mempool_reorg.py` is broken here now. Have to backport a few more PRs to fix this, pls see 3adb3b65729548ed31f83d932073265d8f35453d + 644de37c88bde24bbd3c6352d9059cba03004e3c + 6c02007b668a863528655ebd7722aefdb70c6848

**Review by UdjinM6 — APPROVED** [review]
LGTM, utACK

**Comment by UdjinM6** [comment]
@PastaPastaPasta

File: src/qt/bitcoingui.cpp [hunk 2]
```diff
@@ -401,7 +401,7 @@ void BitcoinGUI::createActions()
     openInfoAction = new QAction(tr("&Information"), this);
     openInfoAction->setStatusTip(tr("Show diagnostic information"));
     openRPCConsoleAction = new QAction(tr("&Debug console"), this);
```
Comment (PastaPastaPasta) [inline src/qt/bitcoingui.cpp:403]: 17096: Wasn't the purpose of this PR to change Debug -> Node
(also probably missing change in debugwindow.ui)

**Review by PastaPastaPasta — CHANGES_REQUESTED** [review]
1 fix needed

**Comment by PastaPastaPasta** [comment]
Also, please rebase on top on latest develop (should fix intermittent test failures)

File: src/qt/bitcoingui.cpp (hunk 2 above)
Comment (UdjinM6) [reply src/qt/bitcoingui.cpp:403]: @PastaPastaPasta pls see https://github.com/dashpay/dash/pull/4896#discussion_r953178746 (+ we changed the title to `Tools` long time ago)

**Review by PastaPastaPasta — APPROVED** [review]
utACK for merging via merge commit

**Comment by thephez** [comment]
@Munkybooty Can you update the PR title to include the conventional commit `backport: ` prefix?

**Pull request merged**